packages/connection-storage/src/connection-storage.ts

```diff
@@ -143,39 +144,41 @@ export class ConnectionStorage {
 
       // While testing, we don't use keychain to store secrets
       if (process.env.COMPASS_E2E_DISABLE_KEYCHAIN_USAGE === 'true') {
-        return await fs.writeFile(
+        await fs.writeFile(
           this.getFilePath(connectionInfo.id),
           JSON.stringify({ connectionInfo }, null, 2),
           'utf-8'
         );
+      } else {
+        const { secrets, connectionInfo: connectionInfoWithoutSecrets } =
+          extractSecrets(connectionInfo);
+        await fs.writeFile(
+          this.getFilePath(connectionInfo.id),
+          JSON.stringify(
+            {
+              connectionInfo: connectionInfoWithoutSecrets,
+            },
+            null,
+            2
+          ),
+          'utf-8'
+        );
+        await keytar
+          .setPassword(
+            this.keytarServiceName,
+            connectionInfo.id,
+            JSON.stringify({ secrets }, null, 2)
+          )
+          .catch((e) => {
+            log.error(
+              mongoLogId(1_001_000_202),
+              'Connection Storage',
+              'Failed to save secrets to keychain',
+              { message: (e as Error).message }
+            );
+          });
```

Comment:
[optional, I know this was already here before] This can just be a regular try/catch, right?

Reply:
added here [`a682345` (#4690)](https://github.com/mongodb-js/compass/pull/4690/commits/a68234595840fb4e7bfc1452027b4dcc3953e1b8)